### DISCUSSION TITLE
Support for sftp

### DISCUSSION THREAD

**Discussion by webolot** [phpactor/phpactor#2544] [General]
Hi,

I'm just installed phpactor for use with Kate editor.

It's working well, for local files, but not for remote files accessed via sftp.

I see this warning on Kate output: `Diagnostic provider "php" errored with "Only "file", "untitled", "phar" schemes are supported, got "sftp"", removing from pool`

It's planned to support sftp on phpactor?

Thank's

**Comment by dantleech** [comment]
Hi, it's not planned currently.

**Reply by webolot** [reply]
Thank's for reply

**Reply by dantleech** [reply]
Converted to a discussion though.

**Comment by webolot** [comment]
After "play" all the morning with my testing environment, I finally came to a conclusion: it would be great if phpactor has support for SFTP ;-)

I tried to mount the ssh filesystem with sshfs, but it's really slow for a large projects (4500 files/folders and 300MB).

Syncronization of project folders with rsync, synthing, or any other software is not an option because a possible 2 way sync problems.

I will follow the development of phpactor to see if it is possible one day.

Thank's!

**Reply by dantleech** [reply]
for context, my current project (on Linux) has 70k files but there are also Mac developers on same code base using Phpactor with Mutagen without so much pain AFAIK.

**Comment by webolot** [comment]
Maybe the problem lies in Kate editor and how it explore the folders tree with sshfs. With the project I mentioned before I have to wait about 80-90 seconds before I can see the file/folder list. 

Using sftp I need to wait 1-2 seconds for the same project.
